yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -656,25 +651,44 @@ export class ProvingOrchestrator {
     const txProvingState = provingState.getTxProvingState(txIndex);
     const publicFunction = txProvingState.getPublicFunctionState(functionIndex);
 
-    // Prove the VM if this is a kernel that requires one
-    if (!KernelTypesWithoutFunctions.has(publicFunction.publicKernelRequest.type)) {
-      // Just sleep for a small amount of time
-      this.deferredProving(
-        provingState,
-        () => sleep(100),
-        () => {
-          logger.debug(`Proven VM for function index ${functionIndex} of tx index ${txIndex}`);
-          this.checkAndEnqueuePublicKernel(provingState, txIndex, functionIndex);
-        },
-      );
+    // If there is a VM request, we need to prove it. Otherwise, continue with the kernel.
+    if (publicFunction.vmRequest) {
+      // This function tries to do AVM proving. If there is a failure, it fakes the proof unless AVM_PROVING_STRICT is defined.
+      // Nothing downstream depends on the AVM proof yet. So having this mode lets us incrementally build the AVM circuit.
+      const doAvmProving = async (signal: AbortSignal) => {
+        const inputs: AvmCircuitInputs = new AvmCircuitInputs(
+          publicFunction.vmRequest!.bytecode,
+          publicFunction.vmRequest!.calldata,
+        );
+        try {
+          return await this.prover.getAvmProof(inputs, signal);
+        } catch (err) {
+          if (process.env.AVM_PROVING_STRICT) {
+            throw err;
+          } else {
+            logger.warn(`Error thrown when proving AVM circuit: ${err}`);
+            logger.warn(`AVM_PROVING_STRICT is off, faking AVM proof and carrying on...`);
+            return { proof: makeEmptyProof(), verificationKey: VerificationKeyData.makeFake() };
```

Comment:
Yes! The public kernels are, AFAIK, currently not taking any VM proof as input. This is because there's still no support for recursive verification for honk proofs in noir. When that changes, this should change. Hopefully we're feature-complete by then!